File: java/baiduads-sdk/src/main/java/com/baidu/dev2/api/sdk/manual/marketingfile/api/MarketingFileService.java
package com.baidu.dev2.api.sdk.manual.marketingfile.api;

import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.invoke.ApiException;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileParams;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileRequest;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileRequestWrapper;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileResponseWrapper;
import com.baidu.dev2.api.sdk.manual.upload.model.MultiUploadOptions;
import com.baidu.dev2.api.sdk.manual.upload.UploadClient;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.HashMap;
import java.util.Map;

public class MarketingFileService {

    public MarketingFileService() {
    }

    public UploadFileResponseWrapper uploadFile(UploadFileRequestWrapper uploadFileRequestWrapper) throws ApiException {
        if (uploadFileRequestWrapper == null) {
            throw new ApiException(400,
                    "Missing the required parameter 'uploadFileRequestWrapper' when calling uploadFile");
        }

        String localVarPath = "/json/sms/service/MarketingFileService/uploadFile";

        Map<String, Object> formParams = new HashMap<String, Object>();
        ObjectNode objectNode = null;
        Integer partParallel = null;
        Integer partSize = null;
        Integer multipartThreshold = null;
        try {
            // set header params
            ApiRequestHeader header = uploadFileRequestWrapper.getHeader();
            Field[] declaredFields = ApiRequestHeader.class.getDeclaredFields();
            for (Field field : declaredFields) {
                if ((Modifier.STATIC & field.getModifiers()) != Modifier.STATIC) {
                    field.setAccessible(true);
                    String name = field.getName();
                    Object value = field.get(header);
                    formParams.put(name, value);
                }
            }

            // set body params
            UploadFileRequest body = uploadFileRequestWrapper.getBody();
            UploadFileParams params = body.getParams();
            ObjectMapper objectMapper = UploadClient.INSTANCE.getApiClient().getObjectMapper();
            objectNode = objectMapper.convertValue(params, ObjectNode.class);
            MultiUploadOptions uploadOptions = body.getUploadOptions();
            if (uploadOptions != null) {
                partParallel = uploadOptions.getPartParallel();
                partSize = uploadOptions.getPartSize();
                multipartThreshold = uploadOptions.getMultipartThreshold();
            }
        } catch (Exception e) {
            throw new ApiException(e);
        }


        return UploadClient.INSTANCE.uploadFile(
                localVarPath,
                formParams,
                uploadFileRequestWrapper.getBody().getFile(),
                objectNode,
                null,
                multipartThreshold,
                partSize,
                partParallel,
                uploadFileRequestWrapper.getBody().getTimeout(),
                new TypeReference<UploadFileResponseWrapper>() {
                }
        );

    }
}

File: python/baiduads-sdk/baiduads_manual/marketingfile/api/marketing_file_service.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Copyright (c) 2022 Baidu.com, Inc. All Rights Reserved
"""

from baiduads_manual.marketingfile.model.upload_file_response_wrapper import UploadFileResponseWrapper
from baiduads_manual.upload.upload_client import UploadClient


class MarketingFileService:
    """MarketingFileService
        please visit:  https://dev2.baidu.com
    """
    def __init__(self):
        self.upload_client = UploadClient()

    def upload_file(self, upload_file_request_wrapper, **kwargs):
        """

        Args:
            upload_file_request_wrapper:UploadFileRequestWrapper
            **kwargs:

        Returns:UploadFileResponseWrapper

        """
        path = "/json/sms/service/MarketingFileService/uploadFile"
        header = upload_file_request_wrapper.header
        request_body = upload_file_request_wrapper.body

        file = request_body.file
        params = request_body.params
        timeout = request_body.timeout

        # form params
        form = []
        # set header param
        attribute_map = header.attribute_map
        data_dict = header.to_dict()
        for k, v in data_dict.items():
            form.append((attribute_map.get(k, k), v))

        upload_options = request_body.uploadOptions
        multipartThreshold = None
        partSize = None
        partParallel = None
        if upload_options is not None:
            multipartThreshold = upload_options.multipartThreshold
            partSize = upload_options.partSize
            partParallel = upload_options.partParallel

        return self.upload_client.upload_file(
            path,
            file,
            form,
            params.__dict__,
            multipartThreshold=multipartThreshold,
            partSize=partSize,
            partParallel=partParallel,
            timeout=timeout,
            returnType=UploadFileResponseWrapper
        )
